packages/client/hmi-client/src/components/models/tera-model-configuration.vue

```diff
@@ -122,56 +148,140 @@ import TabView from 'primevue/tabview';
 import TeraModal from '@/components/widgets/tera-modal.vue';
 import TabPanel from 'primevue/tabpanel';
 import InputText from 'primevue/inputtext';
-import { Model } from '@/types/Types';
-import { ModelConfig } from '@/types/ModelConfig';
+import { ModelConfiguration } from '@/types/Types';
+import { AnyValueMap } from '@/types/common';
+import {
+	createModelConfiguration,
+	updateModelConfiguration
+} from '@/services/model-configurations';
 import { useOpenedWorkflowNodeStore } from '@/stores/opened-workflow-node';
-import { NumericValueMap } from '@/types/common';
+import { ModelOperation } from '@/components/workflow/model-operation';
+
+const openedWorkflowNodeStore = useOpenedWorkflowNodeStore();
 
 const props = defineProps<{
-	model: Model;
 	isEditable: boolean;
-	modelConfigNodeInput?: ModelConfig;
+	modelConfigurations: ModelConfiguration[];
 	calibrationConfig?: boolean;
 }>();
 
-const modelConfigNames = ref(['Config 1']);
+const editableModelConfigs = ref<ModelConfiguration[]>([]);
 
 const selectedModelConfig = ref();
 const fakeExtractions = ref(['Resource 1', 'Resource 2', 'Resource 3']);
 
-const initialValues = ref<NumericValueMap[]>([{}]);
-const parameterValues = ref<NumericValueMap[]>([{}]);
 const openValueConfig = ref(false);
 const cellValueToEdit = ref({ data: {}, field: '', index: 0 });
 
+// const
+
 // Selected columns
 const selectedInitials = ref<string[]>([]);
 const selectedParameters = ref<string[]>([]);
 
-const openedWorkflowNodeStore = useOpenedWorkflowNodeStore();
+// const variableTypes = computed(() => props.model.semantics ? Object.keys(props.model.semantics.ode) : []);
 
-const modelConfiguration = computed(() =>
-	modelConfigNames.value.map((name, i) => ({
-		name,
-		...initialValues.value[i],
-		...parameterValues.value[i]
-	}))
+const configurations = computed(
```

Comment:
What is the expected type on this on?

Reply:
Thanks, I'll cast that to an `object[]` since the names of the keys could vary in the future